### ISSUE TITLE
Deploy Replica Set with arbiter spec using Community Operator v0.7.3

### ISSUE THREAD

**Issue by tzorzan** [mongodb/mongodb-kubernetes-operator#926]
**What did you do to encounter the bug?**
1. Install MongoDB Community Kubernetes Operator and CRDs using [Helm Chart](https://github.com/mongodb/helm-charts/tree/2b3cbe4ff05a67e2f089153542dc9e73511f2713/charts/community-operator) v0.7.3
2. Deploy custom resource of type `MongoDDCommunity`with `arbiter` specification:
```yaml
apiVersion: mongodbcommunity.mongodb.com/v1
kind: MongoDBCommunity
metadata:
  name: test-arbiter-mongodb
spec:
  members: 3
  type: ReplicaSet
  arbiters: 1
  version: "4.2.2"
...
```
3. See the custom resource remaining in _pending_ status
```
▶ k get mdbc -n test                                           
NAME                   PHASE     VERSION
test-arbiter-mongodb   Pending   
```
4. See four pods are created instead of three expected ([documentation](https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/docs/deploy-configure.md#add-arbiters-to-a-replica-set)):
```
▶ k get pod -n test      
NAME                                          READY   STATUS    RESTARTS   AGE
mongodb-kubernetes-operator-748c85557-nmmgz   1/1     Running   0          43m
test-arbiter-mongodb-0                        2/2     Running   0          42m
test-arbiter-mongodb-1                        2/2     Running   0          41m
test-arbiter-mongodb-2                        2/2     Running   0          40m
test-arbiter-mongodb-arb-0                    2/2     Running   0          49m
```
5. See errors in arbiter agent container:
```
[2022-03-17T15:48:54.473+0000] [.info] [src/director/director.go:tracef:793] <test-arbiter-mongodb-arb-0> [15:48:54.473] Running step: 'WaitRsInit' of move 'WaitRsInit'
[2022-03-17T15:48:54.473+0000] [.info] [src/director/director.go:tracef:793] <test-arbiter-mongodb-arb-0> [15:48:54.473] because 
[All the following are true: 
    ['currentState.Up' = true]
    ['currentState.CanRsInit' = false]
    ['desiredState.ReplSetConf' != <nil> ('desiredState.ReplSetConf' = ReplSetConfig{id=test-arbiter-mongodb,version=0,commitmentStatus=false,configsvr=false,protocolVersion=1,forceProtocolVersion=false,writeConcernMajorityJournalDefault=,members={id:0,HostPort:test-arbiter-mongodb-0.test-arbiter-mongodb-svc.test.svc.cluster.local:27017,ArbiterOnly:falsePriority:1,Hidden:false,SecondaryDelaySecs:0,Votes:1,Tags:map[]},{id:1,HostPort:test-arbiter-mongodb-1.test-arbiter-mongodb-svc.test.svc.cluster.local:27017,ArbiterOnly:falsePriority:1,Hidden:false,SecondaryDelaySecs:0,Votes:1,Tags:map[]},{id:2,HostPort:test-arbiter-mongodb-2.test-arbiter-mongodb-svc.test.svc.cluster.local:27017,ArbiterOnly:falsePriority:1,Hidden:false,SecondaryDelaySecs:0,Votes:1,Tags:map[]},{id:100,HostPort:test-arbiter-mongodb-arb-0.test-arbiter-mongodb-arb-svc.test.svc.cluster.local:27017,ArbiterOnly:truePriority:1,Hidden:false,SecondaryDelaySecs:0,Votes:1,Tags:map[]},settings=map[]})]
    ['currentState.ReplSetConf' = <nil>]
]
[2022-03-17T15:48:54.473+0000] [.info] [src/director/director.go:planAndExecute:563] <test-arbiter-mongodb-arb-0> [15:48:54.473] Step=WaitRsInit as part of Move=WaitRsInit in plan failed : <test-arbiter-mongodb-arb-0> [15:48:54.473] Postcondition not yet met for step WaitRsInit because 
['currentState.ReplSetConf' = <nil>].
 Recomputing a plan...
```
6. See operator's log, it can't reconcile:
```
2022-03-17T15:49:52.073Z	INFO	controllers/replica_set_controller.go:140	Reconciling MongoDB	{"ReplicaSet": "test/test-arbiter-mongodb"}
2022-03-17T15:49:52.073Z	DEBUG	controllers/replica_set_controller.go:142	Validating MongoDB.Spec	{"ReplicaSet": "test/test-arbiter-mongodb"}
2022-03-17T15:49:52.073Z	DEBUG	controllers/replica_set_controller.go:151	Ensuring the service exists	{"ReplicaSet": "test/test-arbiter-mongodb"}
2022-03-17T15:49:52.105Z	INFO	controllers/replica_set_controller.go:475	Create/Update operation succeeded	{"ReplicaSet": "test/test-arbiter-mongodb", "operation": "updated"}
2022-03-17T15:49:52.105Z	DEBUG	controllers/replica_set_controller.go:160	Ensuring the service for Arbiters exists	{"ReplicaSet": "test/test-arbiter-mongodb"}
2022-03-17T15:49:52.116Z	INFO	controllers/replica_set_controller.go:475	Create/Update operation succeeded	{"ReplicaSet": "test/test-arbiter-mongodb", "operation": "updated"}
2022-03-17T15:49:52.116Z	INFO	controllers/replica_set_controller.go:371	Creating/Updating AutomationConfig	{"ReplicaSet": "test/test-arbiter-mongodb"}
2022-03-17T15:49:52.133Z	DEBUG	scram/scram.go:166	Credentials have not changed, using credentials stored in: secret/test-arbiter-mongodb-admin-dbroot-scram-credentials
2022-03-17T15:49:52.133Z	DEBUG	controllers/replica_set_controller.go:525	AutomationConfigMembersThisReconciliation	{"mdb.AutomationConfigMembersThisReconciliation()": 3}
2022-03-17T15:49:52.133Z	DEBUG	controllers/replica_set_controller.go:394	Waiting for agents to reach version 1	{"ReplicaSet": "test/test-arbiter-mongodb"}
2022-03-17T15:49:52.133Z	DEBUG	agent/agent_readiness.go:65	The Agent in the Pod 'test-arbiter-mongodb-0' hasn't reached the goal state yet (goal: 1, agent: -1)	{"ReplicaSet": "test/test-arbiter-mongodb"}
2022-03-17T15:49:52.134Z	INFO	controllers/mongodb_status_options.go:110	ReplicaSet is not yet ready, retrying in 10 seconds
```
**What did you expect?**
A Replica Set with 2 data nodes (primary, secondary) and 1 arbiter should be successfully deployed. Custom resource status should be reported as `Ready`and the operator reconcile correctly.

**What happened instead?**
Two different statefulset are created (four total pods) and the operator can't reconcile.

**Operator Information**
 - community-operator v0.7.3 installed from public Helm repo
 - MongoDB Image: `docker.io/mongo:4.2.2`
 - MongoDB Agent Image: `quay.io/mongodb/mongodb-agent:11.12.0.7388-1`

**Kubernetes Cluster Information**
 - AWS EKS with managed nodes
 - 1.21+
 - Image Registry: docker.io and quay (no custom setting)

**Additional context**
Custom resource definition was working correctly with previous chart `mongodb-kubernetes-operator`.

**Comment by rodrigovalin** [comment]
Hi @tzorzan,

I've just pushed a fix for this problem to master. You can try building your own operator from master or wait for a few days for this to be released properly and be installable via Helm.

**Comment by tzorzan** [comment]
Trying to build operator image from master after merge of #941. Maybe I'm missing something in the build&test process, but pods can't start due to a missing mount for ConfigMap:
```
MountVolume.SetUp failed for volume "automation-config" : secret "test-mongodb-config" not found
```
Configmap isn't created actually after CR creation, operator log doesn't mention it as it does in the 0.7.3 version:
```
...
2022-04-13T20:53:24.173Z	DEBUG	controllers/replica_set_controller.go:142	Validating MongoDB.Spec	{"ReplicaSet": "mongodb/test-mongodb"}
2022-04-13T20:53:24.173Z	DEBUG	controllers/replica_set_controller.go:151	Ensuring the service exists	{"ReplicaSet": "mongodb/test-mongodb"}
2022-04-13T20:53:24.184Z	INFO	controllers/replica_set_controller.go:475	Create/Update operation succeeded	{"ReplicaSet": "mongodb/test-mongodb", "operation": "updated"}
2022-04-13T20:53:24.184Z	DEBUG	controllers/replica_set_controller.go:160	Ensuring the service for Arbiters exists	{"ReplicaSet": "mongodb/test-mongodb"}
2022-04-13T20:53:24.192Z	INFO	controllers/replica_set_controller.go:475	Create/Update operation succeeded	{"ReplicaSet": "mongodb/test-mongodb", "operation": "updated"}
2022-04-13T20:53:24.192Z	INFO	controllers/replica_set_controller.go:371	Creating/Updating AutomationConfig	{"ReplicaSet": "mongodb/test-mongodb"}
2022-04-13T20:53:24.211Z	DEBUG	scram/scram.go:166	Credentials have not changed, using credentials stored in: secret/test-mongodb-admin-dbroot-scram-credentials
...
```

The log says now something like:
```
...
2022-04-13T21:10:08.332Z	DEBUG	controllers/replica_set_controller.go:142	Validating MongoDB.Spec	{"ReplicaSet": "test/test-mongodb"}
2022-04-13T21:10:08.332Z	DEBUG	controllers/replica_set_controller.go:151	Ensuring the service exists	{"ReplicaSet": "test/test-mongodb"}
2022-04-13T21:10:08.377Z	INFO	controllers/replica_set_controller.go:459	Create/Update operation succeeded	{"ReplicaSet": "test/test-mongodb", "operation": "updated"}
2022-04-13T21:10:08.378Z	DEBUG	controllers/replica_set_controller.go:408	Scaling up the ReplicaSet, the StatefulSet must be updated first	{"ReplicaSet": "test/test-mongodb"}
2022-04-13T21:10:08.378Z	INFO	controllers/replica_set_controller.go:333	Creating/Updating StatefulSet	{"ReplicaSet": "test/test-mongodb"}
2022-04-13T21:10:08.387Z	INFO	controllers/replica_set_controller.go:338	Creating/Updating StatefulSet for Arbiters	{"ReplicaSet": "test/test-mongodb"}
2022-04-13T21:10:08.413Z	DEBUG	controllers/replica_set_controller.go:348	Ensuring StatefulSet is ready, with type: RollingUpdate	{"ReplicaSet": "test/test-mongodb"}
2022-04-13T21:10:08.413Z	INFO	controllers/mongodb_status_options.go:110	ReplicaSet is not yet ready, retrying in 10 seconds
...
```
not mentioning `AutomationConfig` at all. What I am doing wrong?

Thanky You

**Comment by jinchengsix** [comment]
I had the same problem after deploying 3 nodes and 1 arbiter with master branch（after merge of #941）

**Comment by slaskawi** [comment]
I successfully reproduced this error with the following configuration:

```
---
apiVersion: mongodbcommunity.mongodb.com/v1
kind: MongoDBCommunity
metadata:
  name: test-arbiter-mongodb
spec:
  members: 3
  type: ReplicaSet
  arbiters: 1
  # Note, the version change I made here! Check https://hub.docker.com/_/mongo?tab=tags
  version: "4.2.21"
  security:
    authentication:
      modes: ["SCRAM"]
  users:
    - name: my-user
      db: admin
      passwordSecretRef:
        name: my-user-password
      roles:
        - name: clusterAdmin
          db: admin
        - name: userAdminAnyDatabase
          db: admin
      scramCredentialsSecretName: my-scram
  additionalMongodConfig:
    storage.wiredTiger.engineConfig.journalCompressor: zlib
---
apiVersion: v1
kind: Secret
metadata:
  name: my-user-password
type: Opaque
stringData:
  password: admin
```

Indeed, the `test-arbiter-mongodb` StatefulSet can not start because of the following error:

```
  Warning  FailedMount  27s (x16 over 16m)   kubelet            MountVolume.SetUp failed for volume "automation-config" : secret "test-arbiter-mongodb-config" not found
```

So it seems the main problem was in the order of creating the automation config and the statefulset. A proposed fix might be found here: https://github.com/mongodb/mongodb-kubernetes-operator/pull/1024

**Comment by tzorzan** [comment]
Will try on 0.7.4 release soon. THx

**Comment by otavio** [comment]
I stopped using this operator and moved to bitname meanwhile. Worked for me.

**Comment by ILunie** [comment]
This hasn't solved even at 0.7.6 version
